.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "agent-libs"]
+	path = agent-libs
+	url = https://github.com/KindlingProject/agent-libs.git
```

Comment:
I suggest we write down the branch name explicitly to make things clear.

Reply:
Branching of submodules is allowed, but this is inconsistent with the original design of the submodules, and submodules with specific commits can be used with the current code. Same as libbpf/libbpf bootstrap

Reply:
`https://github.com/libbpf/libbpf-bootstrap/blob/master/.gitmodules`

Reply:
`git submodule` does record a specific commit to allow users to pull the correct repository version. There is no question that the current `.gitmodules` works. AFAIK, tracking a remote branch can make things clear because we can use `git submodule update --remote` to update the code. See [How can I specify a branch/tag when adding a Git submodule?](https://stackoverflow.com/questions/1777854/how-can-i-specify-a-branch-tag-when-adding-a-git-submodule).

It is OK if we use the current `.gitmodules` and make an improvement later.

Reply:
Thank you for your help. That's a great idea. Adding submodule branches is a great way to improve multiplayer collaboration, and it solves a problem I had with using submodule multiplayer collaboration scenarios. Thanks again

Reply:
agent-libs has moved to 'kindling/probe/libs/agent-libs' and add branch mark in .gitmodules